### PR TITLE
Automation subcomponents (no JIRA)

### PR DESCRIPTION
**Pull request by MattPutnam**
Figured out a way to get subcomponents to work. The problem was that Query is an inner type of WebBrowser, so every concrete subclass has its own version of Query, and they're incompatible with each other. This pattern keeps things as Strings for intercommunication, and lets each Component create its Query from that.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [x] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Review by ahaessly — APPROVED** [review]
👍

**Comment by ahaessly** [comment]
just saw this warning in your test run, fyi
[warn] /app/src/test/scala/org/broadinstitute/dsde/firecloud/component/Component.scala:74: implicit conversion method string2QueryString should be enabled
[warn] by making the implicit value scala.language.implicitConversions visible.
[warn] This can be achieved by adding the import clause 'import scala.language.implicitConversions'
[warn] or by setting the compiler option -language:implicitConversions.
[warn] See the Scaladoc for value scala.language.implicitConversions for a discussion
[warn] why the feature should be explicitly enabled.
[warn]   implicit def string2QueryString(s: String) = TestId(s)
[warn]                ^
[warn] one warning found
[warn] javaOptions will be ignored, fork is set to false

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Table.scala [hunk 1]
```diff
@@ -45,48 +42,43 @@ case class Table(private val id: String)(implicit webDriver: WebDriver)
...
   }
 
   def readColumnHeaders: List[String] = {
-    awaitReady()
     readAllText(columnHeaders)
   }
 
-  def clearFilter: Unit = {
-    searchField(filterField).value = ""
-    click on filterButton
```
Comment (breilly2) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Table.scala:82]: I'm not convinced that we don't need to click the filter button and wait. I know that Ursa ran into some timing issues with this that adding these two lines fixed. It's possible that it worked locally but failed when running headless from Jenkins.

**Review by breilly2 — APPROVED** [review]
👍  Thanks for making more improvements! Actually one change request in here but otherwise good to go. If you think the change is not needed, then just keep a close eye on the Jenkins builds. Thanks.

**Pull request merged**